mmdet/apis/train.py

```diff
@@ -84,12 +86,25 @@ def train_detector(model,
 
     # build runner
     optimizer = build_optimizer(model, cfg.optimizer)
-    runner = EpochBasedRunner(
-        model,
-        optimizer=optimizer,
-        work_dir=cfg.work_dir,
-        logger=logger,
-        meta=meta)
+
+    if cfg.get('runner') is None:
+        cfg.runner = {
+            'type': 'EpochBasedRunner',
+            'max_epochs': cfg.total_epochs
+        }
+        warnings.warn(
```

Comment:
During the interface change period, there may be a situation where `total_epochs` and `max_epochs` in the runner exist at the same time, we would better assert this case.